### ISSUE TITLE
[Concerns] Block Dissipation Gone

### ISSUE THREAD

**Issue by SabreMania** [tsunamayo/Starship-EVO#3788]
Okay, so I wanted to make a post or two for a while now regarding suggestions, adding some tools that would make it easier to wire logic and a few other things, however, I'm not going to talk about any of these here, instead i'll address an issue that, at the time of writing this, seems to be the subject of pretty heated debate in the community.

Recently, you (Tsunamayo) removed block-based heat dissipation, because it would be a coding nightmare of some sort, if I am not mistaken. To quote you: >"Yes indeed sorry guys I forgot to mention it, in the name of simplicity I removed heat dissipation stats from hull."<. I'm assuming that the 'sake of simplicity' mentionned here is code-related.

Now, of course, this is an alpha game, I think everyone understands this. At least, everyone relevant. But, this does bring some serious concern. To be quick, it impacts balance in such a way that detailed vessels are at risk. Let me explain:

Previously, if you wanted more heat dissipation, especially on larger vessels, what you did was simply add more blocks. This weighed you down and made you slower and harder to turn, and in exchange you could dissipate more heat. The system needed tweaking, but no block you added ever worked against you. 

Now, in the neovox version, if you want more heat dissipation, regardless of size, you have to increase your surface area. Which means that blocks don't help you directly anymore, which means that there is no more offset for the speed debuff they give.

If it's not clear yet, it means that there's no longer an incentive to building detailed vessels, other than wanting your ship to be pretty, at the cost of performance. Me and many others are very concerned that this, while not direclty and immediately scaring off detailed builders in general, will result in a ship meta where vessels are outfitted with crude, bland interiors, sporting the bare minimum to be called functional, since you want every last one of your blocks to pull their weight. 

Now, i'm not saying 'Add block dissipation back immediately or the game dies', nor am I saying that it's the only way to save the game on the long term, but I am struggling to see alternatives that would actually encourage building interiors, unless you add, for example, blocks that dissipate heat and are clearly meant to be used in interior-building, but that would seem a little weird, or alternatively, blocks that weigh nothing and can be used for interior building, but that would be, in my humble opinion, perhaps a bit counter-productive.

In short, me and a number of other people are afraid that this will go down the Starmade road, where, for the sake of efficiency, ships were as bland as can be in order to maximise all of the factors that made a good ship good. Schema, the Starmade developper, tried all sorts of ways to fix that problem, but they all failed. Seeing this game solve that problem with heat and block-based heat dissipation was a great surprise, but now that it's gone, I fear for the worst.

**Comment by Servomotor** [comment]
As an agreeable person with impeccable taste, I 100% agree with SabreMania.

I think this sums of my concern. Something I said in the official discord. 

"If more dense/detailed ships are going to be at a disadvantage, people will not build them. If people stop building interesting ships that showcase what the game can do, it will not be as enticing or interesting to new players, and may chase off old ones. 

Nerfing dense/detailed ships is literally a bad business model."

**Comment by TIKIRobo** [comment]
One thing that Tsuna said was that the heat dissipation per block wasn't scaling right (so making your interiors wholly out of smaller blocks would make your ship very op) so yes as long as Tsuna can fix that then yes there should be heat dissipation by volume AND surface area.

**Comment by ProPeach** [comment]
I've talked about my feelings on this in the other thread, but I'll link them here for reference - https://github.com/tsunamayo/Starship-EVO/issues/3784#issuecomment-846480038

In short, I thought the combination of surface area and hull was great, which we can improve further by simplifying the different scales of hull to have the same stats, and give smaller ships a helping hand with a surface area to volume ratio based heat modifier .

**Comment by Garrett-C** [comment]
I do think this discussion brings up a good point. Part of the aim of systems in the game is to make it so that you can build a functional ship that still has a cool interior. With blocks inside the ship no longer assisting in heat dissipation it means every interior block is now just dead weight. Adding rooms adds mass with no quantifiable gain to ship performance. In theory this means ships without interiors will outperform those with them.

**Comment by Kaiser-Indrasil** [comment]
> I do think this discussion brings up a good point. Part of the aim of systems in the game is to make it so that you can build a functional ship that still has a cool interior. With blocks inside the ship no longer assisting in heat dissipation it means every interior block is now just dead weight. Adding rooms adds mass with no quantifiable gain to ship performance. In theory this means ships without interiors will outperform those with them.

Quite on the contrary. Since the heat dissipation is calculated based on the surface area, the more surface you have inside the ship, the better. It directly translates to

More fleshed-out interior -> more heat dissipation.

**Comment by ProPeach** [comment]
> 
> Quite on the contrary. Since the heat dissipation is calculated based on the surface area, the more surface you have inside the ship, the better. It directly translates to
> 
> More fleshed-out interior -> more heat dissipation.

This isn't correct, the surface area that matters is the *exterior* surface area. It follows irl rules, any heat radiated by your chair will just be absorbed by your interior walls etc, so no heat gain there

**Comment by Kaiser-Indrasil** [comment]
Oh, so it's smarter than I assumed. Interesting! Thanks for clarifying that

**Comment by tsunamayo** [comment]
Sorry for the lack of update, I am changing place so I havent resumed normal work pace atm...
@SabreMania thanks for the feedback, it matches other builder concerns. I will put back hull dissipation in the next update. No need to be that dramatic ;) - this is a preview of an experimental build, so if I can afford to try things.
In my test hull heat only kicks in with very large ship. As I gave a massive reduction to heat generation with large system I was under the impression that they are gonna be fine after some system rebuild.

I wanted to get rid of it for simplification sake (not code related). Having two sources of heat dissipation is a bit complex (but not the end of the world).
But I agree I also need to give a bit more incentive for interior making - that was the goal of the heat mechanism at the first place.

To give more color my game balance reasoning is:
 - how can I make smaller build more powerful relative to larger build? => By making smaller system more powerful compared to bigger one volume wise. Also by adding the surface heat mechanism.
 - how can I give intensive for not building only with mini system as those are more powerful for a given volume? My answer was by making larger system more heat efficient. Smaller ship dont care (thanks to the surface heat mechanism !), but that is important for larger one.

Now please keep in mind that I will put them back but with a lower heat per block generated:
**I do need to nerf heat dissipation on large / very large build!** Because those ship are supposed to be build with 2m, 4m system (which of course they are not currently). Otherwise that would mean that my system heat generation scaling is not effective. That being said if your ship is too stacked inside so you dont have room to replace system please tell me, I still need to calibrate things.

I hope it makes sense.

As always what is really useful for me is data, so if you have trouble with a particular design send it my way I will check it (even if I dont have the time atm, sorry for that ;))

Cheers

**Comment by Garrett-C** [comment]
> Sorry for the lack of update, I am changing place so I havent resumed normal work pace atm...
> @SabreMania thanks for the feedback, it matches other builder concerns. I will put back hull dissipation in the next update. No need to be that dramatic ;) - this is a preview of an experimental build, so if I can afford to try things.
> In my test hull heat only kicks in with very large ship. As I gave a massive reduction to heat generation with large system I was under the impression that they are gonna be fine after some system rebuild.
> 
> I wanted to get rid of it for simplification sake (not code related). Having two sources of heat dissipation is a bit complex (but not the end of the world).
> But I agree I also need to give a bit more incentive for interior making - that was the goal of the heat mechanism at the first place.
> 
> To give more color my game balance reasoning is:
> 
> * how can I make smaller build more powerful relative to larger build? => By making smaller system more powerful compared to bigger one volume wise. Also by adding the surface heat mechanism.
> * how can I give intensive for not building only with mini system as those are more powerful for a given volume? My answer was by making larger system more heat efficient. Smaller ship dont care (thanks to the surface heat mechanism !), but that is important for larger one.
> 
> Now please keep in mind that I will put them back but with a lower heat per block generated:
> **I do need to nerf heat dissipation on large / very large build!** Because those ship are supposed to be build with 2m, 4m system (which of course they are not currently). Otherwise that would mean that my system heat generation scaling is not effective. That being said if your ship is too stacked inside so you dont have room to replace system please tell me, I still need to calibrate things.
> 
> I hope it makes sense.
> 
> As always what is really useful for me is data, so if you have trouble with a particular design send it my way I will check it (even if I dont have the time atm, sorry for that ;))
> 
> Cheers

That makes a lot of sense I think. Your logic for balancing things is definitely good. 

I guess part of the issue you have is the game hasn't been very optimised for super big ships until recently so I imagine it was sort of tough to really get a handle on the balancing side. Hopefully with the new novoxel tech you will get some more examples to play with in the future. 

Yeah reducing the amount of cooling from hull blocks seems reasonable. Still incentivises interiors while also incentivising using bigger systems. 

Good luck with the move. Hope things are going smoothly and you get settled in comfortably.

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo I agree with @Garrett-C , your idea is very good indeed. I love when you explain things to the letter, it really helps.

We would have to fine-tune the exact numbers behind the heat dissipation calculations in the months to come, but the principle itself holds really well, at least in theory.

And - what especially calms my mind about this - it will incentivise a meta with fleshed-out interiors, fewer bigger system blocks instead of tons of smaller ones, while ensuring that the smaller ships are still perfectly viable!
That is some great game design right here.

Have you thought about a name for your proprietary block system? Since we can't call it 'voxel' anymore, we with a few guys on Discord started referring to it as 'novox'.
We used to call them 'neovox' but since it misrepresented it by implying that it's a derivative of the voxel tech (and that pissed off Womble ;) ), we switched to the current 'novox' name.
But since 'novox' doesn't sound catchy enough, have you thought of a more suitable name?

Oh, and I've heard that moving homes absolutely sucks, so hang in there!

**Comment by tsunamayo** [comment]
Actually I think I prefer neovox. It also make sense as it is what comes after voxel. I think avorion had coined a term, boxel if I recall.
I could go ahead and call my tech neovox...

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo It's funny, cos I came up with that name when getting hyped up about the first new update with the guys on Discord, but only sort of as a meme. It's nice to see it coming around 😊

**Comment by tsunamayo** [comment]
Okay the experimental build is dropping soon with the heat fix. Thanks all!
